In [1]:
################ Chi-Square Test ################
import pandas as pd

#Ho: All Proportions are equal
#Ha:not all proportions are equal

In [2]:
Bahaman = pd.read_excel("D:\\upendra\\Data Science Training\\hypothesis testing\\Bahaman.xlsx")
Bahaman



Defective     Country
0            0       India
1            0       India
2            0       India
3            0       India
4            1       India
..         ...         ...
795          0  Bangladesh
796          0  Bangladesh
797          1  Bangladesh
798          0  Bangladesh
799          0  Bangladesh

[800 rows x 2 columns]

In [4]:
Bahaman.Country.unique()

array(['India', 'China', 'Srilanka', 'Bangladesh'], dtype=object)

In [5]:
count = pd.crosstab(Bahaman["Defective"], Bahaman["Country"])
count


Country    Bangladesh  China  India  Srilanka
Defective                                    
0                 183    179    175       178
1                  17     21     25        22

In [7]:
import scipy 
from scipy import stats
Chisquares_results = scipy.stats.chi2_contingency(count)
Chisquares_results

(1.7243932538050184,
 0.6315243037546223,
 3,
 array([[178.75, 178.75, 178.75, 178.75],
        [ 21.25,  21.25,  21.25,  21.25]]))

In [8]:
Chi_square = [['Test Statistic', 'p-value'], [Chisquares_results[0], Chisquares_results[1]]]

In [ ]:
Chi_square

In [9]:
Chi_square

[['Test Statistic', 'p-value'], [1.7243932538050184, 0.6315243037546223]]

#### since .6315>.05 hence H0 will be accepted ie all proportions are equal